lib/graphql/models/definition_helpers/associations.rb

```diff
@@ -41,7 +41,7 @@ def self.resolve_has_one_type(reflection)
           graph_types = valid_types.map { |t| GraphQL::Models.get_graphql_type(t) }.compact
 
           GraphQL::UnionType.define do
-            name "#{model_type.name}#{reflection.foreign_type.classify}"
+            name "#{model_type.class_name}#{reflection.foreign_type.classify}"
```

Comment:
Perhaps instead, this code could be modified to do some sanitization on the name, stripping it down to only alphanumeric characters:
```ruby
model_type.name.gsub(/[^A-Za-z0-9_]+/, '')
```